### PR TITLE
plugins return maps instead of arrays

### PR DESCRIPTION
**Pull request by giladgray**
- add `documentalist.objectify` function to convert array to map
- update theme and plugins
- namespace plugin interfaces so the nav is more legible

### CODE REVIEW

File: src/plugins/typescript.ts [hunk 1]
```diff
@@ -9,37 +9,40 @@ import tsdoc, { IJsDocTags } from "ts-quick-docs";
...
-export interface IInterfaceEntry extends IDocEntry {
+export interface ITsInterfaceEntry extends ITsDocEntry {
     extends?: string[];
-    properties: IPropertyEntry[];
+    properties: ITsPropertyEntry[];
 }
 
-export interface ITypescriptPluginData {
-    ts: IInterfaceEntry[];
+export interface ITsPluginData {
```
Comment (themadcreator) [inline src/plugins/typescript.ts]: I think all these sub interfaces are fine with the shortened "Ts" name, but this one in particular should match the name of the plugin so `ITypescriptPluginData`

File: theme/index.pug [hunk 2]
```diff
@@ -1,4 +1,4 @@
-- ts.sort((a, b) => a.name.localeCompare(b.name))
+- var toc = Object.keys(ts).sort((a, b) => a.localeCompare(b))
```
Comment (themadcreator) [inline theme/index.pug]: heh, unobjectified then

File: src/documentalist.ts [hunk 3]
```diff
@@ -172,6 +178,13 @@ export class Documentalist<T> implements IApi<T> {
         return documentation;
     }
 
+    public objectify<T>(array: T[], getKey: (item: T) => string) {
```
Comment (themadcreator) [inline src/documentalist.ts]: This doesn't really belong on the API, this is a stateless utility method.

File: theme/index.pug (hunk 2 above)
Comment (giladgray) [inline theme/index.pug]: i don't understand your words?

File: src/documentalist.ts (hunk 3 above)
Comment (giladgray) [inline src/documentalist.ts]: that is true but we don't have a place for such things

File: theme/index.pug (hunk 2 above)
Comment (themadcreator) [inline theme/index.pug]: just that you changed this array to an object in this PR and now you're converting back to an array.

**Comment by themadcreator** [comment]
Fine, but we need to move that objectify method eventually

File: theme/index.pug (hunk 2 above)
Comment (giladgray) [inline theme/index.pug]: well sure we need a sorted array for the table of contents. often both forms are needed, so i guess it's a tradeoff either way.

Comment (themadcreator) [inline theme/index.pug]: yeah im fine with these changes, just found it amusing

**Pull request merged**